openmicroscopy/openmicroscopy
# Map annotation web

This PR adds Map Annotations to web.
Testing includes these areas:

 - Check that edits on the client are sync'd to the server with no data loss or overwriting (during copy, paste, edit, delete etc).
 - Check in DB that only a single map annotation per user per PDI is created.
 - Permissions testing - canAnnotate() allows adding of Map annotations, canEdit() allows editing of someone else's map annotation. TODO: removal of whole annotation.
 - Also good to compare clients and suggestions to improve UI & usability.

--no-rebase

![screen shot 2015-01-16 at 10 00 58](https://cloud.githubusercontent.com/assets/900055/5774516/aca1743a-9d66-11e4-840d-4ed1235faea8.png)
